src/core/ddsc/src/dds_qos.c

```diff
@@ -335,6 +335,151 @@ void dds_qset_ignorelocal (dds_qos_t * __restrict qos, dds_ignorelocal_kind_t ig
   qos->present |= QP_CYCLONE_IGNORELOCAL;
 }
 
+static void dds_qprop_init (dds_qos_t * qos)
+{
+  if (!(qos->present & QP_PROPERTY_LIST))
+  {
+    qos->property.value.n = 0;
+    qos->property.value.props = NULL;
+    qos->property.binary_value.n = 0;
+    qos->property.binary_value.props = NULL;
+    qos->present |= QP_PROPERTY_LIST;
+  }
+}
+
+static bool dds_qprop_get_index (const dds_qos_t * qos, const char * name, uint32_t * index)
+{
+  bool found = false;
+  if (qos == NULL || name == NULL || index == NULL || !(qos->present & QP_PROPERTY_LIST))
+    return false;
+  for (uint32_t i = 0; !found && i < qos->property.value.n; i++)
+  {
+    if (strcmp (qos->property.value.props[i].name, name) == 0)
+    {
+      *index = i;
+      found = true;
+    }
```

Comment:
Stylistically, I much prefer an early-out in such cases, eliminating the state and — more importantly — the mental state involved in avoiding the early-out. Styles are quite personal and early-outs can cause discussion almost as fierce as indentation or emacs-vs-vi, so all I'm asking is that you consider it in the future.

Reply:
I agree to use early-out here because the function is only a few lines. I fixed this.